portal/server/utils/csv-utils.js

```diff
@@ -0,0 +1,137 @@
+const ExcelJS = require('exceljs');
+const csv = require('csv-parser');
+const { Readable } = require('stream');
+
+const columnIndexToExcelColumn = (index) => {
+  let result = '';
+  let indexTracker = index;
+  while (indexTracker >= 0) {
+    result = String.fromCharCode((index % 26) + 65) + result;
+    indexTracker = Math.floor(index / 26) - 1;
+  }
+  return result;
+};
+
+const parseXlsxToCsvArrays = (worksheet) => {
+  const csvData = [];
+  const csvDataWithCellAddresses = [];
+
+  let firstRow = true;
+  worksheet.eachRow((row) => {
+    const rowData = [];
+    const rowDataWithCellAddresses = [];
+    row.eachCell({ includeEmpty: true }, (cell) => {
+      rowData.push(cell.value);
+      if (firstRow) {
+        rowDataWithCellAddresses.push(`${cell.value}`);
+      } else {
+        rowDataWithCellAddresses.push(`${cell.value}-${cell.address}`);
+      }
+    });
+    firstRow = false;
+    csvData.push(rowData.join(','));
+    csvDataWithCellAddresses.push(rowDataWithCellAddresses.join(','));
+  });
+  return { csvData, csvDataWithCellAddresses };
+};
+
+const xlsxBasedCsvToJsonPromise = async (csvDataWithCellAddresses) => {
+  const csvStream = new Readable({
+    read() {
+      for (const line of csvDataWithCellAddresses) {
+        this.push(`${line}\n`); // Add newline to simulate line-by-line reading
+      }
+      this.push(null); // End the stream
+    },
+  });
+  return new Promise((resolve, reject) => {
+    try {
+      const jsonParsedData = [];
+      csvStream
+        .pipe(
+          csv({
+            mapHeaders: ({ header }) => header.toLowerCase().replace(/\s/g, ' ').trim(),
+            mapValues: ({ value }) => {
+              const lastDashIndex = value.lastIndexOf('-');
+              const cellValue = value.substring(0, lastDashIndex);
+              const cellAddress = value.substring(lastDashIndex + 1);
+
+              // Split the cell address into column and row
+              const [column, row] = cellAddress.match(/([A-Z]+)(\d+)/).slice(1);
```

Comment:
Will move this regex to somewhere else, also I need to improve it as sonarcloud flagged it as being vulnerable to super-linear runtime due to backtracking